File: java/src/main/java/org/caseontology/ProfileCritic.java
package org.caseontology;

/** Logical ProfileCritic surface. Construction findings stay on InvestigationBuilder. */
public final class ProfileCritic {
    public final ProfileContract contract;

    public ProfileCritic(ProfileContract contract) {
        this.contract = contract;
    }

    public static ProfileCritic forProfile(String profileId) {
        return new ProfileCritic(ProfileContract.load(profileId));
    }
}

File: csharp/CaseUco/ProfileCritic.cs
// Logical ProfileCritic surface — construction findings stay on InvestigationBuilder.

namespace CaseUco
{
    public sealed class ProfileCritic
    {
        public ProfileContract Contract { get; }

        public ProfileCritic(ProfileContract contract)
        {
            Contract = contract;
        }

        public static ProfileCritic ForProfile(string profileId)
        {
            return new ProfileCritic(ProfileContract.Load(profileId));
        }
    }
}
